### PR TITLE
[SCHEMA] Add suffix term files

### PR DESCRIPTION
**Pull request by tsalo**
Closes None.

Changes proposed:
- Add a new folder within the schema of YAML files describing valid BIDS file suffixes.
- Render suffix tables with a macro instead of hardcoded tables.

To do:

- [x] Add functions to schemacode to render suffix description tables.
- [x] Remove hardcoded tables (after review).
- [x] Determine final structure for suffix schema.
     - Currently, I just have `description`, `name`, `unit`, `minValue`, and `maxValue`.
     - Should we include "valid extensions", or is that dependent on data type?
     - Should we include a deprecated flag, or is that dependent on data type?
     - One key difference from metadata is that the `name` field in suffixes is used to store the full name for the data stored in files with that suffix, while `name` in metadata files is used to differentiate between the actual field and the name of the file, so we can have multiple definitions for the same field by storing them in separate files.

### CODE REVIEW

File: src/schema/suffixes/RB1COR.yaml
Comment (tsalo) [inline src/schema/suffixes/RB1COR.yaml:4]: This could use a real description.

File: src/schema/suffixes/TB1DAM.yaml
Comment (tsalo) [inline src/schema/suffixes/TB1DAM.yaml:4]: This could use a real description.

File: src/schema/suffixes/TB1EPI.yaml
Comment (tsalo) [inline src/schema/suffixes/TB1EPI.yaml:4]: This could use a real description.

File: src/schema/suffixes/TB1RFM.yaml
Comment (tsalo) [inline src/schema/suffixes/TB1RFM.yaml:4]: This could use a real description.

File: src/schema/suffixes/TB1SRGE.yaml
Comment (tsalo) [inline src/schema/suffixes/TB1SRGE.yaml:4]: This could use a real description.

File: src/schema/suffixes/TB1TFL.yaml
Comment (tsalo) [inline src/schema/suffixes/TB1TFL.yaml:4]: This could use a real description.

File: src/schema/suffixes/VFA.yaml
Comment (tsalo) [inline src/schema/suffixes/VFA.yaml:4]: This could use a real description.

**Comment by effigies** [comment]
@tsalo Is this still WIP or are you interested in a review?

**Comment by tsalo** [comment]
I'd like to tinker with it a bit before a full review once #774 is merged, since there will definitely be a few merge conflicts to deal with, but after that I'd love any and all feedback.

**Comment by tsalo** [comment]
@effigies Okay, I think it's ready for review.

File: src/schema/suffixes/bold.yaml
Comment (effigies) [inline src/schema/suffixes/bold.yaml:2]: This table is an odd one out in not having full names. It's fine since all have short descriptions with the full names, but wanted to make sure this was the goal.

File: tools/schemacode/schema.py
Comment (tsalo) [reply tools/schemacode/schema.py:8]: Sorry about that. I have VSCode automatically `isort`ing my files now, and sometimes unexpected changes slip through.

File: src/04-modality-specific-files/01-magnetic-resonance-imaging-data.md
Comment (tsalo) [reply src/04-modality-specific-files/01-magnetic-resonance-imaging-data.md:209]: Okay, I can move this to a separate PR.

File: src/schema/suffixes/bold.yaml
Comment (tsalo) [reply src/schema/suffixes/bold.yaml]: I just changed it to the full name. Thanks!

File: src/04-modality-specific-files/01-magnetic-resonance-imaging-data.md
Comment (tsalo) [reply src/04-modality-specific-files/01-magnetic-resonance-imaging-data.md]: Now in #830.

File: src/schema/suffixes/bold.yaml
Comment (effigies) [inline src/schema/suffixes/bold.yaml:2]: I think you've been using image

```suggestion
name: Blood-Oxygen-Level Dependent image
```

File: src/schema/suffixes/cbv.yaml
Comment (effigies) [inline src/schema/suffixes/cbv.yaml:2]:
```suggestion
name: Cerebral blood volume image
```

File: src/schema/suffixes/phase.yaml
Comment (effigies) [inline src/schema/suffixes/phase.yaml:2]:
```suggestion
name: Phase image
```

**Review by effigies — APPROVED** [review]
LGTM. Small changes.

File: src/schema/suffixes/_template.yaml
Comment (sappelhoff) [inline src/schema/suffixes/_template.yaml:3]: is this template supposed to be committed?

Comment (tsalo) [reply src/schema/suffixes/_template.yaml:3]: I like to share the template so it's easier to just copy over to a new file as needed. I don't think it causes any problems (at least it doesn't seem to have with the metadata files), but I can remove it if there's an issue.

File: src/schema/suffixes/eeg.yaml
Comment (sappelhoff) [inline src/schema/suffixes/eeg.yaml:4]: Am I right in assuming that some of the created YAML files are not yet integrated in the spec text in this PR?

**Review by sappelhoff — APPROVED** [review]
apart from my two comments LGTM

going from "map" to "image" was probably a conscious decision, as was going from abbreviations to full names.

File: src/schema/suffixes/eeg.yaml
Comment (tsalo) [reply src/schema/suffixes/eeg.yaml:4]: I think the only part of the specification with suffix tables is the MRI page, so none of the other suffixes, including `eeg`, are rendered in the text. To do so, we'd need to introduce new tables rather than just replace existing ones, I believe.

File: src/schema/suffixes/_template.yaml
Comment (sappelhoff) [reply src/schema/suffixes/_template.yaml:3]: okay, that's fine - just wanted to avoid an accidental commit

File: src/schema/suffixes/eeg.yaml
Comment (sappelhoff) [reply src/schema/suffixes/eeg.yaml:4]: :+1:

**Comment by tsalo** [comment]
Thanks for reviewing @effigies and @sappelhoff! Now that we have two approvals, I can trigger the five-day waiting period and merge next Thursday if no one raises any concerns.

**Review by sappelhoff — CHANGES_REQUESTED** [review]
just a note that we still need to remove the hardcoded tables before merging.

(I think you can merge starting from today, as it's been 5 days)

**Review by sappelhoff — APPROVED** [review]

**Comment by tsalo** [comment]
Okay, the hardcoded tables are gone.

**Comment by sappelhoff** [comment]
Thanks @tsalo

**Pull request merged**
